src/Storages/System/StorageSystemViews.cpp

```diff
@@ -0,0 +1,68 @@
+#include <Storages/System/StorageSystemViews.h>
+#include <DataTypes/DataTypeString.h>
+#include <Access/ContextAccess.h>
+#include <Interpreters/Context.h>
+#include <Interpreters/QueryViewsLog.h>
+#include <DataTypes/DataTypeEnum.h>
+#include <Storages/StorageMaterializedView.h>
+#include <Storages/LiveView/StorageLiveView.h>
+
+namespace DB
+{
+
+class Context;
+
+NamesAndTypesList StorageSystemViews::getNamesAndTypes()
+{
+    auto view_type_datatype = std::make_shared<DataTypeEnum8>(DataTypeEnum8::Values{
+        {"Default", static_cast<Int8>(QueryViewsLogElement::ViewType::DEFAULT)},
+        {"Materialized", static_cast<Int8>(QueryViewsLogElement::ViewType::MATERIALIZED)},
+        {"Live", static_cast<Int8>(QueryViewsLogElement::ViewType::LIVE)}});
+
+    return {
```

Comment:
@kssenii We need better column names. Now they can be easily confused to each other. Also we need to highlight that `table` is just the main table in the sequence of JOINs.

Reply:
At first column names were: `database`, `table`, `view_database`, `view_table`, but it was suggested be Maxim [in the comment above](https://github.com/ClickHouse/ClickHouse/pull/28082#discussion_r694887321) that they should be as they are now:
```
I am not sure if naming is good enough.
From my perspective better name for table is system.views.
For columns table system.views should have same naming as system.tables.
database - database of view
name - name of view
table - view table
table_database - view table database.
```

Reply:
Ok.

Reply:
What about:
`main_dependency_table`, `main_dependency_database`?

Reply:
@alexey-milovidov I am not sure if this table system.views is actually needed, maybe better to add something to system.tables using Map data type.